src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -3260,6 +4225,115 @@ type QueryType {
         """
         startTransitTripId: String @deprecated(reason: "Not implemented in OTP2")
     ): Plan @async
+
+    """
+    Plan (itinerary) search that follows
+    [GraphQL Cursor Connections Specification](https://relay.dev/graphql/connections.htm).
+    """
+    planConnection(
+        """
+        Datetime of the search. It's possible to either define the earliest departure time
+        or the latest arrival time. By default, earliest departure time is set as now.
+        """
+        dateTime: PlanDateTimeInput
+
+        """
+        Duration of the search window. This either starts at the defined earliest departure
+        time or ends at the latest arrival time. If this is not provided, a reasonable
+        search window is automatically generated. When searching for earlier or later itineraries
+        with paging, this search window is no longer used and the new window will be based
+        on how many suggestions were returned in the previous search. The new search window can be
+        shorter or longer than the original search window. Note, itineraries are returned faster
+        with a smaller search window and search window limitation is done mainly for performance reasons.
+
+        Setting this parameter makes especially sense if the transportation network is as sparse or dense
+        in the whole itinerary search area. Otherwise, letting the system decide what is the search window
+        is in combination of using paging can lead to better performance and to getting a more consistent
+        number of itineraries in each search.
+        """
+        searchWindow: Duration
+
+        """
+        Maximum number of itineraries that will be returned. It's possible that no itineraries are
+        found, paging for earlier or later itineraries can yield to results in that case. The returned
+        itineraries are either closest to the defined earliest departure time or to the latest arrival
+        time. It can make sense to search for more itineraries that what is immediately needed if there
+        is a possibility that more itineraries are used later on (limiting the number of returned itineraries
+        does not affect the performance by a lot but can affect the network bandwidth usage).
+        """
+        numberOfItineraries: Int
```

Comment:
What did we decide about this being required/having a default value? I can't remember anymore if we made some decision or not.

Reply:
or did we decide to not limit it by default at all and just limit through the search window?

Reply:
The `numberOfItineraries` should have a default value (set in code). It is used to crop the searchwindow AFTER the search is done so setting it too low will result in poor performance. If a client can only show 3 results then 3 is ok, if many results are allowed then setting it to 20/50 is fine. Note! this only reduces the amount of data retuned - it does not limit the search.